examples/chef/common/chef-fan-control-manager.cpp

```diff
@@ -99,98 +113,205 @@ Status ChefFanControlManager::HandleStep(StepDirectionEnum aDirection, bool aWra
+        break;
+    }
+
+    case FanControl::Attributes::SpeedSetting::Id: {
+        ChipLogProgress(NotSpecified, "ChefFanControlManager::HandleFanControlAttributeChange  SpeedSetting");
+
+        // DataModel::Nullable<uint8_t> speedSetting = static_cast<DataModel::Nullable<uint8_t>>(*value);
+        DataModel::Nullable<uint8_t> speedSetting = NumericAttributeTraits<DataModel::Nullable<uint8_t>>::StorageToWorking(*value);
+        SetSpeedCurrent(speedSetting.IsNull() ? 0 : speedSetting.Value());
+        // Determine if the speed change should also change the fan mode
+        FanControl::Attributes::FanMode::Set(mEndpoint, SpeedToFanMode(mSpeedCurrent));
+        break;
+    }
+
+    case FanControl::Attributes::FanMode::Id: {
+        ChipLogProgress(NotSpecified, "ChefFanControlManager::HandleFanControlAttributeChange  FanMode");
+
+        FanControl::FanModeEnum fanMode = static_cast<FanControl::FanModeEnum>(*value);
+        FanModeWriteCallback(fanMode);
+        break;
+    }
+
+    default: {
+        break;
+    }
+    }
+}
 
-    switch (aPath.mAttributeId)
+FanControl::FanModeEnum ChefFanControlManager::SpeedToFanMode(uint8_t speed)
+{
+    if (speed == 0)
+    {
+        return FanControl::FanModeEnum::kOff;
+    }
+    else if (speed <= FAN_MODE_LOW_UPPER_BOUND)
+    {
+        return FanControl::FanModeEnum::kLow;
+    }
+    else if (speed <= FAN_MODE_MEDIUM_UPPER_BOUND)
+    {
+        return FanControl::FanModeEnum::kMedium;
+    }
+    else if (speed <= FAN_MODE_HIGH_UPPER_BOUND)
     {
-    case SpeedSetting::Id: {
-        Nullable<uint8_t> newSpeedSetting;
-        ReturnErrorOnFailure(aDecoder.Decode(newSpeedSetting));
+        return FanControl::FanModeEnum::kHigh;
+    }
+    return FanControl::FanModeEnum::kUnknownEnumValue;
+}
 
-        // Ensure new speed is in bounds
+void ChefFanControlManager::SetPercentCurrent(uint8_t aNewPercentCurrent)
+{
+    if (aNewPercentCurrent != mPercentCurrent)
+    {
+        ChipLogDetail(NotSpecified, "ChefFanControlManager::SetPercentCurrent: %d", aNewPercentCurrent);
+        mPercentCurrent = aNewPercentCurrent;
+        Status status   = FanControl::Attributes::PercentCurrent::Set(mEndpoint, mPercentCurrent);
+        if (status != Status::Success)
         {
-            uint8_t maxSpeedSetting                    = 0;
-            Protocols::InteractionModel::Status status = SpeedMax::Get(mEndpoint, &maxSpeedSetting);
-            VerifyOrReturnError(status == Protocols::InteractionModel::Status::Success, CHIP_IM_GLOBAL_STATUS(Failure));
-
-            if (!newSpeedSetting.IsNull() && newSpeedSetting.Value() > maxSpeedSetting)
-            {
-                return CHIP_IM_GLOBAL_STATUS(ConstraintError);
-            }
+            ChipLogError(NotSpecified, "ChefFanControlManager::SetPercentCurrent: failed to set PercentCurrent attribute: %d",
+                         to_underlying(status));
+            return;
```

Comment:
this has a return and SetSpeedCurrent does not. I would make it consistent (maybe decrease indent in both and no return, since we use this status for log only)

Reply:
Right, made them consistent. Thanks!